include/psen_scan_v2/diagnostics.h

```diff
@@ -0,0 +1,184 @@
+  { Dc::INT_COM_PRB, "Internal communication problem." },
+  { Dc::GENERIC_ERR, "Generic Error." },
+  { Dc::DISP_COM_PRB, "Display communication problem." },
+  { Dc::TEMP_MEAS_PROB, "Temperature measurement problem." },
+  { Dc::ENCOD_OOR, "Encoder: Out of range." },
+  { Dc::EDM2_ERR, "Error in the External Device Monitoring (EDM2_ERR)." },
+  { Dc::EDM1_ERR, "Error in the External Device Monitoring (EDM1_ERR)." },
+  { Dc::CONF_ERR, "Configuration Error." },
+  { Dc::OUT_OF_RANGE_ERR, "Out of range error." },
+  { Dc::TEMP_RANGE_ERR, "Temperature out of range." },
+  { Dc::ENCODER_GENERIC_ERR, "Encoder: Generic error." },
+  { Dc::_, "Unexpected error" } \
+};
+
+// clang-format off
+  #define REV(arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8) arg8, arg7, arg6, arg5, arg4, arg3, arg2, arg1
+
+  static constexpr std::array<std::array<DiagnosticCode, 8>, 9> error_bits{{
+  //Bit7                 Bit6              Bit5              Bit4              Bit3              Bit2                  Bit1                   Bit0
+  { REV(Dc::OSSD1_OC,    Dc::OSSD_SHRT_C,  Dc::OSSD_INTEGR,  Dc::INT,          Dc::INT,          Dc::INT,              Dc::INT,               Dc::INT) },
+  { REV(Dc::WIN_CLN_AL,  Dc::POWER_SUPPLY, Dc::NETW_PRB,     Dc::DUST_CRC_FL,  Dc::INT,          Dc::INT,              Dc::_,                 Dc::OSSD2_OVERCUR) },
+  { REV(Dc::MEAS_PROB,   Dc::INT,          Dc::INT,          Dc::INT,          Dc::INCOHERENCE,  Dc::ZONE_INVAL_TRANS, Dc::ZONE_INVALID_CONF, Dc::WIN_CLN_WARN) },
+  { REV(Dc::INT_COM_PRB, Dc::INT,          Dc::INT,          Dc::GENERIC_ERR,  Dc::DISP_COM_PRB, Dc::INT,              Dc::INT,               Dc::TEMP_MEAS_PROB) },
+  { REV(Dc::ENCOD_OOR,   Dc::_,            Dc::_,            Dc::EDM2_ERR,     Dc::EDM1_ERR,     Dc::CONF_ERR,         Dc::OUT_OF_RANGE_ERR,  Dc::TEMP_RANGE_ERR) },
+  { REV(Dc::_,           Dc::_,            Dc::_,            Dc::_,            Dc::_,            Dc::_,                Dc::_,                 Dc::ENCODER_GENERIC_ERR) },
+  { REV(Dc::UNUSED,      Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,           Dc::UNUSED,            Dc::UNUSED) },
+  { REV(Dc::UNUSED,      Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,           Dc::UNUSED,            Dc::UNUSED) },
+  { REV(Dc::UNUSED,      Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,       Dc::UNUSED,           Dc::UNUSED,            Dc::UNUSED) },
+  }};
+// clang-format on
+
+class ErrorLocation
+{
+public:
+  using byteLocation = size_t;
+  using bitLocation = size_t;
+  ErrorLocation(byteLocation byte, bitLocation bit) : byte_(byte), bit_(bit){};
+  inline byteLocation getByte() const
+  {
+    return byte_;
+  };
+  inline bitLocation getBit() const
+  {
+    return bit_;
+  };
+
+private:
+  byteLocation byte_;
+  bitLocation bit_;
+};
+
+class MonitoringFrameDiagnosticMessage
```

Comment:
It should be possible to make this class `constexpr` because it gets created and then it does not really change anymore.